app/controllers/materials_controller.rb

```diff
@@ -0,0 +1,51 @@
+class MaterialsController < ApplicationController
+
+  def index
+    @materials = policy_scope(Material).all
+  end
+
+  def show
+    @material = Material.find(params[:id])
+    authorize @material
+  end
+
+  def new
+    @material = Material.new
+    authorize @material
+  end
+
+  def create
+    @material = Material.new(material_params)
+    if @material.save
+      redirect_to materials_path
+    else
+      render :new
+    end
+    authorize @material
+  end
+
+  def edit
+    @material = Material.find(params[:id])
+    authorize @material
+  end
+
+  def update
+    @material = Material.find(params[:id])
+    @material.update(material_params)
```

Comment:
Maybe we could use a if statement and render edit if failed like for the save ?

Reply:
Edit not finished yet